engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -56,67 +55,80 @@ class ValidateActor()
     * 2.) Raw inputs and it's coercion
     * 3.) Workflow Options
     * 4.) RuntimeAttributes w.r.t. a backend
+    *
     * @return The result of the validation as instance of `ValidationResult`
     */
-  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = {
-    (for {
-      namespaceWithWorkflow <- Future(NamespaceWithWorkflow.load(wdlSource))
-      validatedInputs <- Future(validateInputs(workflowInputs, namespaceWithWorkflow))
-      validatedWorkflowOptions <- Future(validateWorkflowOptions(workflowOptions, CromwellBackend.backend()))
-      validatedRuntimeAttrs <- Future(validateRuntimeAttributes(namespaceWithWorkflow))
-    } yield ValidationSuccess(namespaceWithWorkflow, validatedInputs, validatedWorkflowOptions, validatedRuntimeAttrs)) recover {
-      case reason => ValidationFailure(reason)
+  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = Future {
```

Comment:
I will remove the future (agree on Miguel's comment about control) but I still think it's a mistake to have a singleton as the first step in a web-service implementation. In CCC there are people working on a component for creating workflows executions on the fly based on federated data sets and it may be a bottleneck for us and by definition is a bad idea to have a singleton behind a web-service.